wagtail/wagtail
# Rework existing page editor shortcuts behaviour to be easier to build on

Moves the existing, non-Draftail, keyboard shortcuts to one central include with the goal of making it easier to register new shortcuts and providing a modal that shows all available shortcuts.

## Changed behaviour

* Changed preview shortcut from `mod+p` (as this overrides the print in browsers) and used `alt+p` (option+p in mac).
* When a keyboard shortcut is triggered for the save / preview pane, it will also focus on the button (more suitable for accessibility), this makes no real difference for the save shortcut.
* They keyboard shortcuts will not activate when a modal is open, this causes confusion and is unhelpful to the user. We can revisit this if we find a suitable use case for this.
* Modals (dialogs) now dispatch an event when they open & close, not documented intentionally as we may want to adopt a different naming convention in the near future.

## How to test

* Load up any edit page and confirm you can use `cmd+s` on Mac or `ctrl+s` on Windows to make the save page trigger.
* Load up any edit page and confirm you can use `option+p` on Mac or `alt+p` on Windows to toggle the Preview side pain
* Load up any edit page and confirm you can use `command+option+m` on Mac or `ctrl+alt+m` on Windows while on a specific field to show the comment field (note: comments must be toggled on).

## Details

- Move Mousetrap package to npm package and not static asset.
- Adopt a data attribute driven keyboard shortcut declaration approach, along with a way to register shortcuts via dispatching an event. Not documented intentionally as we may change this before we finalise this approach in future work.
- Pull out keyboard shortcut JS to generic include file
- Sdd unit tests for some basic JS behaviour
- relates to https://github.com/wagtail/wagtail/issues/3949
